src/formatDate.js

```diff
@@ -8,7 +8,57 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromValue = fromFormat[fromFormat.length - 1];
+
+  const toValue = toFormat[toFormat.length - 1];
+
+  const newDate = date.split(fromValue);
+
+  const newArray = [];
+
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = newDate[i];
+  }
+
+  const indexOfYear = fromFormat.includes('YYYY')
+    ? fromFormat.indexOf('YYYY')
+    : fromFormat.indexOf('YY');
+
+  if (!('YYYY' in dateObject)) {
+    if (newDate[indexOfYear] >= 30) {
+      dateObject.YYYY = '19' + newDate[indexOfYear];
+    }
+
+    if (newDate[indexOfYear] < 30) {
+      dateObject.YYYY = '20' + newDate[indexOfYear];
+    }
+  }
+
+  if ('YYYY' in dateObject) {
+    dateObject.YY = newDate[indexOfYear].slice(2);
```

Comment:
The assignment of `YY` assumes that `newDate[indexOfYear]` is always a valid year string. Add validation to ensure this assumption is correct.